### PR TITLE
USHIFT-494: go1.18

### PR DESCRIPTION
**Pull request by pmtk**
- updates go version `go.mod`
- makes necessary changes to dev-env doc & scripts and ansible playbook to work around rhel8.6 not having go1.18 in repos - **these should be reverted when we start using RHEL 8.7**

### CODE REVIEW

File: ansible/roles/setup-microshift-host/tasks/main.yml [hunk 1]
```diff
@@ -7,6 +7,13 @@
     state: present
     update_cache: true
 
+- name: install golang
+  ansible.builtin.shell: |
```
Comment (pmtk) [inline ansible/roles/setup-microshift-host/tasks/main.yml:11]: @sjug I know this is a little janky, but hopefully after upgrading to RHEL 8.7 we'll be able to revert this

File: ansible/roles/setup-microshift-host/tasks/main.yml [hunk 2]
```diff
@@ -7,6 +7,13 @@
...
+    export GO_VER=1.18.7
+    curl -L -o go${GO_VER}.linux-amd64.tar.gz https://go.dev/dl/go${GO_VER}.linux-amd64.tar.gz && \
+    sudo rm -rf /usr/local/go && sudo tar -C /usr/local -xzf go${GO_VER}.linux-amd64.tar.gz
+    echo 'export PATH=$PATH:/usr/local/go/bin' >> ~/.bashrc && source ~/.bashrc
```
Comment (ggiguash) [inline ansible/roles/setup-microshift-host/tasks/main.yml:15]: If we're not installing 'golang' package, I would rather create a symlink from /usr/local/bin/go pointing to the new package. Let's not create a dependency on custom environment.

File: docs/devenv_rhel8.md [hunk 3]
```diff
@@ -55,8 +55,10 @@ Run the following commands to configure SUDO, upgrade the system, install basic
 ```bash
 echo -e 'microshift\tALL=(ALL)\tNOPASSWD: ALL' | sudo tee /etc/sudoers.d/microshift
 sudo dnf update -y
-sudo dnf install -y git cockpit make golang selinux-policy-devel rpm-build bash-completion
+sudo dnf install -y git cockpit make gcc selinux-policy-devel rpm-build bash-completion
 sudo systemctl enable --now cockpit.socket
+export GO_VER=1.18.7; curl -L -o go${GO_VER}.linux-amd64.tar.gz https://go.dev/dl/go${GO_VER}.linux-amd64.tar.gz && sudo rm -rf /usr/local/go && sudo tar -C /usr/local -xzf go${GO_VER}.linux-amd64.tar.gz
+echo 'export PATH=$PATH:/usr/local/go/bin' >> ~/.bashrc && source ~/.bashrc
```
Comment (ggiguash) [inline docs/devenv_rhel8.md:61]: If we're not installing 'golang' package, I would rather create a symlink from /usr/local/bin/go pointing to the new package. Let's not create a dependency on custom environment.

File: docs/devenv_rhel8.md [hunk 4]
```diff
@@ -55,8 +55,10 @@ Run the following commands to configure SUDO, upgrade the system, install basic
 ```bash
 echo -e 'microshift\tALL=(ALL)\tNOPASSWD: ALL' | sudo tee /etc/sudoers.d/microshift
 sudo dnf update -y
-sudo dnf install -y git cockpit make golang selinux-policy-devel rpm-build bash-completion
+sudo dnf install -y git cockpit make gcc selinux-policy-devel rpm-build bash-completion
 sudo systemctl enable --now cockpit.socket
+export GO_VER=1.18.7; curl -L -o go${GO_VER}.linux-amd64.tar.gz https://go.dev/dl/go${GO_VER}.linux-amd64.tar.gz && sudo rm -rf /usr/local/go && sudo tar -C /usr/local -xzf go${GO_VER}.linux-amd64.tar.gz
```
Comment (ggiguash) [inline docs/devenv_rhel8.md:60]: Can we install to /usr/local/go-${GO_VER} directory? 
Also, we need to delete the .tar.gz file afterwards.

**Review by ggiguash — CHANGES_REQUESTED** [review]

File: docs/devenv_rhel8.md [hunk 5]
```diff
@@ -78,6 +80,7 @@ The artifact of the build is the `microshift` executable file located in the `_o
 ### RPM Packages
 Run make command with the `rpm` or `srpm` argument in the top-level directory. 
 ```bash
+sed -i 's,^BuildRequires: golang,# &,' ./packaging/rpm/microshift.spec
```
Comment (ggiguash) [inline docs/devenv_rhel8.md:83]: I would suggest permanently commenting out this requirement until we get the right RPM package. 
Can we do this in the .spec and leave an appropriate comment?

**Review by ggiguash — CHANGES_REQUESTED** [review]

File: docs/devenv_rhel8.md (hunk 5 above)
Comment (pmtk) [reply docs/devenv_rhel8.md:83]: But this works in CI - image used is based on rhel8 with go 1.18. This change is only needed for local dev env

Comment (ggiguash) [reply docs/devenv_rhel8.md:83]: I know, but still it requires unmanaged edits of source files, which is messy. 
If we do this, we would need to explain why we're doing this. I think we could avoid all the mess by temporary commenting out this statement if it does not have other consequences.

File: docs/devenv_rhel8.md (hunk 3 above)
Comment (pmtk) [reply docs/devenv_rhel8.md]: Done

File: docs/devenv_rhel8.md (hunk 4 above)
Comment (pmtk) [reply docs/devenv_rhel8.md]: Done

File: docs/devenv_rhel8.md (hunk 5 above)
Comment (pmtk) [reply docs/devenv_rhel8.md]: Done

**Comment by dhellmann** [comment]
/lgtm

/override ci/prow/e2e-openshift-conformance-sig-arch
/override ci/prow/e2e-openshift-conformance-sig-storage
/override ci/prow/e2e-openshift-conformance-sig-auth
/override ci/prow/e2e-openshift-conformance-sig-network
/override ci/prow/e2e-openshift-conformance-sig-cli
/override ci/prow/e2e-openshift-conformance-sig-apps
/override ci/prow/e2e-openshift-conformance-sig-node
/override ci/prow/e2e-openshift-conformance-sig-instrumentation
/override ci/prow/e2e-rpm-install
/override ci/prow/e2e-openshift-conformance-sig-scheduling
/override ci/prow/e2e-openshift-conformance-sig-api-machinery

**Pull request merged**